File: VNC.CodeAnalysis Test Cases/TestCasesVB/Quality/MagicNumbersInMath.vb
Public Class MagicNumbersInMath
    Public Function fun(g As Integer) As Single
        Dim size As Integer = 10000
        g += 23456
        g += size
        Return CSng((g / 10))
    End Function

    Public Function updateRate(rate As Decimal) As Decimal
        Return rate / 0.2345D
    End Function

    Public Function updateRateM(rateM As Decimal) As Decimal
        Dim basis As Decimal = New Decimal(2345, 0, 0, False, 4)
        Return rateM / basis
    End Function
End Class

File: CodeAnalysis Test Cases/TestCasesCS/Quality/MagicNumbersInMath.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSTestCases.Quality
{
    public class MagicNumbersInMath
    {
        public float fun(int g)
        {
            int size = 10000;
            g+=23456;//bad code. magic number 23456 is used.
            g+=size;
            return g/10;
        }

        public decimal updateRate(decimal rate)
        {
            return rate / 0.2345M;
        }

        public decimal updateRateM(decimal rateM)
        {
            decimal basis = 0.2345M;
            return rateM/basis;
        }
    }
}
